azurerm/resource_arm_application_gateway.go

```diff
@@ -1255,6 +1270,7 @@ func expandApplicationGatewayBackendHTTPSettings(d *schema.ResourceData, gateway
 		port := int32(v["port"].(int))
 		protocol := v["protocol"].(string)
 		cookieBasedAffinity := v["cookie_based_affinity"].(string)
+		hostName := v["host_name"].(string)
```

Comment:
this is an optional value so we should only be setting it when its not empty?

Reply:
What if we'd like to `unset` this field through empty string ?

Reply:
@metacpp I've tested setting and unsetting both parameters and it works well. This PR also includes an error checking for pick_host_name_from_backend_address == true and host_name set.